src/coreclr/jit/importer.cpp

```diff
@@ -18912,15 +18912,15 @@ void Compiler::impMakeDiscretionaryInlineObservations(InlineInfo* pInlineInfo, I
         }
     }
 
-    // Check if callee is a generic method when its caller is not
-    bool calleeIsGeneric = (info.compMethodInfo->args.callConv & CORINFO_CALLCONV_GENERIC) ||
-                           (info.compClassAttr & CORINFO_FLG_SHAREDINST);
-    bool callerIsGeneric = (rootCompiler->info.compMethodInfo->args.callConv & CORINFO_CALLCONV_GENERIC) ||
-                           (rootCompiler->info.compClassAttr & CORINFO_FLG_SHAREDINST);
-
-    if (!callerIsGeneric && calleeIsGeneric)
+    if ((info.compMethodInfo->args.callConv & CORINFO_CALLCONV_GENERIC) ||
+        (info.compClassAttr & CORINFO_FLG_SHAREDINST))
+    {
+        inlineResult->Note(InlineObservation::CALLEE_GENERIC);
+    }
+    else if ((rootCompiler->info.compMethodInfo->args.callConv & CORINFO_CALLCONV_GENERIC) ||
+             (rootCompiler->info.compClassAttr & CORINFO_FLG_SHAREDINST))
     {
-        inlineResult->Note(InlineObservation::CALLSITE_GENERIC_FROM_NONGENERIC);
+        inlineResult->Note(InlineObservation::CALLSITE_GENERIC);
```

Comment:
Does this check catch generic methods in non-generic classes?

Reply:
Yes, via `compMethodInfo->args.callConv & CORINFO_CALLCONV_GENERIC`